src/components/PostDetails.tsx

```diff
@@ -1,117 +1,126 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { deleteComment, getComments, postComment } from '../api/comments';
+import { Comment, CommentData } from '../types/Comment';
+import { CommentsList } from './CommentsList';
+
+type Props = {
+  post: Post,
+};
+
+export const PostDetails: React.FC<Props> = ({
+  post,
+}) => {
+  const {
+    id,
+    title,
+    body,
+  } = post;
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isError, setIsError] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isFormOpen, setIsFormOpen] = useState<boolean>(false);
+  const [isFormLoading, setIsFormLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    setIsFormOpen(false);
+    setIsError(false);
+
+    getComments(id)
+      .then(setComments)
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  }, [post]);
```

Comment:
The useEffect hook has 'post' as a dependency, but 'id' is the actual data used inside the hook. Using 'post' as a dependency might cause unnecessary re-renders if other properties of 'post' change. Consider using 'id' as a dependency instead.